### PR TITLE
add authorization to faucet client

### PR DESCRIPTION
**Pull request by ngkuru**

### CODE REVIEW

File: crates/aptos-rest-client/src/faucet.rs [hunk 1]
```diff
@@ -39,20 +41,32 @@ impl FaucetClient {
                 // versioned API however, so we just set it to `/`.
                 .version_path_base("/".to_string())
                 .unwrap(),
+            token: None,
         }
     }
 
+    // Set auth token.
+    pub fn with_token (mut self, token: String) -> Self {
```
Comment (banool) [inline crates/aptos-rest-client/src/faucet.rs:49]: Nit: `with_auth_token`

File: crates/aptos-rest-client/src/faucet.rs [hunk 2]
```diff
@@ -83,10 +97,15 @@ impl FaucetClient {
...
         // returning.
-        let response = self
+        let mut request = self
             .inner
             .post(url)
-            .header("content-length", 0)
+            .header("content-length", 0);
+        if let Some(token) = &self.token {
+            request = request.header("Authorization", format!("Bearer {}", token));
+        }
```
Comment (banool) [inline crates/aptos-rest-client/src/faucet.rs:106]: we have this twice, maybe we need another function like `build_request`.

**Review by banool — APPROVED** [review]
Looks great! Make sure that it works and then feel free to land, pending second stamp.

**Review by gregnazario — APPROVED** [review]

**Comment by gregnazario** [comment]
please run `scripts/rust_lint.sh`

**Pull request merged**